guides/source/active_record_migrations.md

```diff
@@ -623,6 +623,33 @@ NOTE: Active Record only supports single column foreign keys. `execute` and
 `structure.sql` are required to use composite foreign keys. See
 [Schema Dumping and You](#schema-dumping-and-you).
 
+### Composite Primary Keys
+
+Sometimes a single column's value isn't enough to uniquely identify every row
+of a table, but a combination of two or more columns *does* uniquely identify
+it. This can be the case when using a legacy database schema without a single
+`id` column as a primary key, or when altering schemas for sharding or
+multitenancy.
+
+You can create a table with a composite primary key by passing the
+`:primary_key` option to `create_table` with an array value:
+
+```ruby
+class CreateProducts < ActiveRecord::Migration[7.1]
+  def change
+    create_table :products, primary_key: [:customer_id, :product_sku] do |t|
+      t.integer :customer_id
+      t.string :product_sku
+      t.text :description
+    end
+  end
+end
+```
+
+INFO: Tables with composite primary keys require passing array values rather
+than integer IDs to many methods. See also the [Active Record Querying][]
```

Comment:
The link isn't rendering properly for me right now. I think we need to specify the URL below, ie.
```md
[Active Record Querying]: active_record_querying.html
```

Reply:
I copied the syntax from elsewhere in the guide. But that might be too optimistic. The Active Record Associations guide is linked that way, though.

Reply:
I think that one works because it's defined below 😄 
https://github.com/rails/rails/blob/44bd6e7acaa91dd59c260beea6863a49b84f6039/guides/source/active_record_migrations.md?plain=1#L579

(You can tophat the guides by doing `cd guides; rake guides:generate:html ONLY=active_record_migrations`, and then previewing the html file it generates!)

Reply:
D'oh! Okay, I'll fix that quick.